lib/msal-node/test/client/ManagedIdentityApplication.spec.ts

```diff
@@ -194,6 +199,103 @@ describe("ManagedIdentityApplication unit tests", () => {
+                    systemAssignedConfig
+                );
+            });
+
+            test("acquires a token", async () => {
+                expect(ManagedIdentityTestUtils.isAzureArc()).toBe(true);
+
+                const networkManagedIdentityResult: AuthenticationResult =
+                    await managedIdentityApplication.acquireToken(
+                        managedIdentityRequestParams
+                    );
+                expect(networkManagedIdentityResult.fromCache).toBe(false);
+
+                expect(networkManagedIdentityResult.accessToken).toEqual(
+                    DEFAULT_SYSTEM_ASSIGNED_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
+                );
+            });
+
+            test("returns an already acquired token from the cache", async () => {
+                expect(ManagedIdentityTestUtils.isAzureArc()).toBe(true);
+
+                const networkManagedIdentityResult: AuthenticationResult =
+                    await managedIdentityApplication.acquireToken({
+                        resource: MANAGED_IDENTITY_RESOURCE,
+                    });
+                expect(networkManagedIdentityResult.fromCache).toBe(false);
+
+                expect(networkManagedIdentityResult.accessToken).toEqual(
+                    DEFAULT_SYSTEM_ASSIGNED_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
+                );
+
+                const cachedManagedIdentityResult: AuthenticationResult =
+                    await managedIdentityApplication.acquireToken({
+                        resource: MANAGED_IDENTITY_RESOURCE,
+                    });
+                expect(cachedManagedIdentityResult.fromCache).toBe(true);
+                expect(cachedManagedIdentityResult.accessToken).toEqual(
+                    DEFAULT_SYSTEM_ASSIGNED_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
+                );
+            });
+
+            test("attempts to acquire a token, a 401 and WWW-Authenticate header are returned form the azure arc managed identity, then retries the network request with the WWW-Authenticate header", async () => {
+                expect(ManagedIdentityTestUtils.isAzureArc()).toBe(true);
+
+                const managedIdentityApplicationAzureArc401: ManagedIdentityApplication =
+                    new ManagedIdentityApplication({
+                        system: {
+                            networkClient:
+                                ManagedIdentityTestUtils.getManagedIdentityNetworkAzure401Client() as INetworkModule,
+                            // managedIdentityIdParams will be omitted for system assigned
+                        },
+                    });
+
+                const sendGetRequestAsyncSpy = sinon.spy(
+                    Azure401CustomHttpClient.prototype,
+                    <any>"sendGetRequestAsync"
+                );
+
+                try {
+                    // this request will fail because of the way it's mocked
```

Comment:
Add another test where you get the token successfully after the header is added to the request.

Reply:
Also add tests to verify the error response when the auth header cannot be parsed or is empty or the file path is not in the response.

Reply:
Regarding your first comment: that's not possible because of the way the http client is mocked. I can't modify the managed identity's http client after it's created. That's why this test is set up the way it is. I check if the get request was called twice, and ensure that the second request has the auth header.